include/sentry.h

```diff
@@ -1408,45 +1447,65 @@ SENTRY_EXPERIMENTAL_API sentry_span_t *sentry_transaction_start_child(
  * finishing the Span means it will be discarded, and will not be sent to
  * sentry. `sentry_value_null` will be returned if the child Span could not be
  * created.
+ *
+ * The Span passed in must have a read lock to be thread-safe. This increases
+ * the number of references pointing to the passed-in Span's parent Transaction.
+ *
+ * The returned value is not thread-safe. Users are expected to ensure that
+ * appropriate locking mechanisms are implemented over the Span if it needs
+ * to be mutated across threads. Methods operating on the Span will
+ * mention what kind of expectations they carry if they ned to mutate or access
+ * the object in a thread-safe way.
  */
 SENTRY_EXPERIMENTAL_API sentry_span_t *sentry_span_start_child(
     sentry_span_t *parent, char *operation, char *description);
 
 /**
  * Finishes a Span.
  *
- * `root_transaction` is either the parent Transaction of the Span, or
- * the ancestor Transaction of the Span if the Span is not a direct descendant
- * of a Transaction.
- *
  * This takes ownership of `span`, as child Spans must always occur within the
  * total duration of a parent Span and cannot outlive their parent Spans.
+ *
+ * The Span passed in must have an exclusive write lock to be thread-safe. This
+ * will mutate the Span's parent Transaction, and therefore also requires an
+ * exclusive write lock on that Transaction as well.
```

Comment:
this is actually pretty difficult to communicate. i'm on the fence about this being a positive side-effect of this change; the end user just needs to know which transaction the span belongs to, and to ensure that there's a write lock on that transaction when `sentry_span_finish` is invoked. a lot of this is very implicit.

Reply:
```
Takes ownership of the span, mutates the transaction the span was associated with,
which will require locking if that transaction is used from multiple threads.
```

I agree that it is now harder for users to know which transaction needs the locking.  We haven't made things worse though: with the old API they **had** to keep track of this, now they only have to keep track of this if they want to use a single transaction in multiple threads.